### PR TITLE
update for using multiple visual for collada_urdf

### PR DESCRIPTION
**Pull request by YoheiKakiuchi**
I added code for using multiple visual for converting urdf to collada.
If urdf link has just one visual, there is no difference with former result.
urdf link has more than two visual, first visual would be added with former manner.
Next visuals would be added as transformed geometry.

### CODE REVIEW

File: collada_urdf/src/collada_urdf.cpp [hunk 1]
```diff
@@ -1148,24 +1148,66 @@ class ColladaWriter : public daeErrorHandler
...
-            pinstmat->setSymbol("mat0");
+            bool write_visual = false;
+            if ( !!plink->visual ) {
+              if (plink->visual_groups.size() > 0) {
+                std::map<std::string, boost::shared_ptr<std::vector<boost::shared_ptr<urdf::Visual > > > >::const_iterator def_group
+                  = plink->visual_groups.find("default");
+                if (def_group != plink->visual_groups.end()) {
+                  std::cerr << "found default visual group" << std::endl;
+                  if (def_group->second->size() > 1) {
+                    std::cerr << "visual group size = " << def_group->second->size() << std::endl;
```
Comment (isucan) [inline collada_urdf/src/collada_urdf.cpp:1162]: Could we remove this?

File: collada_urdf/src/collada_urdf.cpp [hunk 2]
```diff
@@ -1148,24 +1148,66 @@ class ColladaWriter : public daeErrorHandler
...
+                      pinstmat->setTarget(xsAnyURI(*pdomgeom, string("#")+geomid+string("_mat")));
+                      pinstmat->setSymbol("mat0");
+                      write_visual = true;
+                    }
+                  }
+                }
+              }
+            }
+            if (!write_visual) {
+              std::cerr << "use default visual 1" << std::endl;
```
Comment (isucan) [inline collada_urdf/src/collada_urdf.cpp:1192]: and this

**Comment by isucan** [comment]
This is great, thank you. Do you just take the "default" group and ignore the rest? 

Can we do the same for the collision tags?

File: collada_urdf/src/collada_urdf.cpp (hunk 1 above)
Comment (YoheiKakiuchi) [inline collada_urdf/src/collada_urdf.cpp:1162]: I have left some error messages. They were not needed. Please remove lines start with std::cerr .

**Comment by YoheiKakiuchi** [comment]
> Do you just take the "default" group and ignore the rest?

Yes. I don't know exact usage visual(collision) group. It seems to have detailed visual(collision) and simple one, and will change depend on situation. I have no idea to do that in collada format. 

> Can we do the same for the collision tags?

Yes. Now exported collada file has just visual description (same as collision), so we should choose visual tag or collision tag for exporting.

**Comment by isucan** [comment]
Is it not possible to export both collision and visual in collada? (I am not too familiar with the collada format either). How does one choose visual / collision tags now?

**Comment by YoheiKakiuchi** [comment]
Now in urdf_to_collada, if visual tag exists, visual tag is used as geometry(collision/visual) in collada, and ignore collision tag.
If visual tag does not exist, collision tag is used. If both visual and collision doesn't exist, there is nothing to do (link has no visual/collision). 

Collision geometries extension is described at http://openrave.org/docs/latest_stable/collada_robot_extensions/
Now, urdf_to_collada doesn't support this extension. I think it is future work.

**Pull request merged**